src/transformers/models/fuyu/image_processing_fuyu.py

```diff
@@ -0,0 +1,252 @@
+import math
+from typing import List, Union
+
+import numpy as np
+
+from ...image_processing_utils import BaseImageProcessor
+from ...image_transforms import (
+    normalize,
+    pad,
+    resize,
+)
+from ...image_utils import to_numpy_array
+from ...utils import is_torch_available, is_vision_available, logging, requires_backends
+
+
+if is_vision_available():
+    import PIL
+
+if is_torch_available():
+    import torch
+
+logger = logging.get_logger(__name__)
+
+
+class FuyuImageProcessor(BaseImageProcessor):
```

Comment:
wait, this PR gets an approval with the current image processor? We're lucky Amy isn't seeing this :D

On a more serious note, the image processor needs a big rework to make it compliant with all our existing image processors (e.g. the size argument needs to be a dictionary by default, e.g. size = {"height": 1080, "width": 1920}, arguments like `do_resize` and `do_pad` need to be added, a separate preprocess function needs to be implemented, etc.

See also `Pix2StructImageProcessor` as a reference example

Reply:
For sure 🙌 but it’s a time sensitive release gotta merge now and refactor after

Reply:
The processor needs to be refactored as well and the config for default arguments as well + no TestMixin

Reply:
Might be a few but there won’t be a release before this is fixed so should be alright I hope.

Reply:
Yeah the ImageProcessor has to be reworked, will be easier to do once we have the model up and running as well. I'll continue working on it and make it aligned with the rest :)

Reply:
Ok :)